cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/Artifacts.java

```diff
@@ -45,7 +45,14 @@ public static String getFileName(ArtifactId artifactId) {
   public static Type getConfigType(Class<? extends Application> appClass) {
     TypeToken<?> configType = TypeToken.of(appClass).resolveType(Application.class.getTypeParameters()[0]);
     if (Reflections.isResolved(configType.getType())) {
-      return configType.getType();
+      // Default the type to Config.class if the resolved type is not subclass of Config.
+      // It normally won't happen, unless someone generate the bytecode directly.
+      // This is for Scala. If the user don't specify any type parameter, Scala will automatically insert
+      // "scala.runtime.Nothing$" as the type parameter.
+      if (Config.class.isAssignableFrom(configType.getRawType())) {
```

Comment:
Oh ok, I thought it would also catch it if someone makes their Application parameterized by something silly like `Integer`, and this case would catch that (I didn't think about the Java interface)
But since you say its only happening when someone writes the bytecode directly, it wont happen in that config.